pkg/api/index.go

```diff
@@ -110,9 +110,12 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 
 			if include.Type == "dashboard" && include.AddToNav {
 				link := &dtos.NavLink{
-					Url:  hs.Cfg.AppSubURL + "/dashboard/db/" + include.Slug,
+					Url:  hs.Cfg.AppSubURL + "/d/" + hs.PluginManager.GetDashboardUIDBySlug(c.OrgId, include.Slug),
```

Comment:
Think this is still my main concern I think. Feels a bit wrong. I would still like to see support for using `include.UID` if that's available. Then potentially fallback to this solution, but would be nicer if the `include` was pre-populated with the `UID`. Maybe moving parts of this logic into `hs.PluginManager.GetEnabledPlugins` as a first step would allow the plugin manager to keep some implementation details internal?

Reply:
Unless I'm missing something, `Include` can not be pre-populated with `UIDs` at all, since `Includes` is a singleton, while `UIDs` might be different per organisation. Cloning `Includes` for each organisation only because the one and only field there is different sounds like a waste to me.

To keep things local - we can still rework it to cache `(org, slug) => UID` relations lazily inside the plugin manager after making a database query once per plugin on the first page load. I am not sure if PluginManager will be able to invalidate such a cache on its own, though.

Reply:
I opened a [draft PR](#36789) of an alternative related to some of my early thoughts. Please read the description of it.